### DISCUSSION TITLE
firstOrCreate()->first() ?

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#33333] [General] [answered]
Guys I mistakenly put `first()`  after `firstOrCreate`, But it returns the first record of the table not the current record, This is excepted ?

```php
Person::query()->firstOrCreate(['name' => 'alex']);

Person::query()->firstOrCreate(['name' => 'john'])->first(); // returns alex!
```

**Comment by netpok** [comment, marked as answer]
It is, you're calling a non existing method on a model so it will be passed to the builder. It is basically the same as calling:
- `Person::first()` or
- `(new Person())->first()`
